// @iterable/react-native-sdk/js/IterableInAppClasses.js
'use strict';
/**
 * `show` to show the in-app otherwise `skip` to skip.
 */
var IterableInAppShowResponse;
(function (IterableInAppShowResponse) {
    IterableInAppShowResponse[IterableInAppShowResponse["show"] = 0] = "show";
    IterableInAppShowResponse[IterableInAppShowResponse["skip"] = 1] = "skip";
})(IterableInAppShowResponse || (IterableInAppShowResponse = {}));
/**
 * `immediate` will try to display the in-app automatically immediately
 * `event` is used for Push to in-app
 * `never` will not display the in-app automatically via the SDK
 */
var IterableInAppTriggerType;
(function (IterableInAppTriggerType) {
    IterableInAppTriggerType[IterableInAppTriggerType["immediate"] = 0] = "immediate";
    IterableInAppTriggerType[IterableInAppTriggerType["event"] = 1] = "event";
    IterableInAppTriggerType[IterableInAppTriggerType["never"] = 2] = "never";
})(IterableInAppTriggerType || (IterableInAppTriggerType = {}));
class IterableInAppTrigger {
    type;
    constructor(type) {
        this.type = type;
    }
    static fromDict(dict) {
        const type = dict["type"];
        return new IterableInAppTrigger(type);
    }
}
var IterableInAppContentType;
(function (IterableInAppContentType) {
    IterableInAppContentType[IterableInAppContentType["html"] = 0] = "html";
    IterableInAppContentType[IterableInAppContentType["alert"] = 1] = "alert";
    IterableInAppContentType[IterableInAppContentType["banner"] = 2] = "banner";
})(IterableInAppContentType || (IterableInAppContentType = {}));
var IterableInAppLocation;
(function (IterableInAppLocation) {
    IterableInAppLocation[IterableInAppLocation["inApp"] = 0] = "inApp";
    IterableInAppLocation[IterableInAppLocation["inbox"] = 1] = "inbox";
})(IterableInAppLocation || (IterableInAppLocation = {}));
var IterableInAppCloseSource;
(function (IterableInAppCloseSource) {
    IterableInAppCloseSource[IterableInAppCloseSource["back"] = 0] = "back";
    IterableInAppCloseSource[IterableInAppCloseSource["link"] = 1] = "link";
    IterableInAppCloseSource[IterableInAppCloseSource["unknown"] = 100] = "unknown";
})(IterableInAppCloseSource || (IterableInAppCloseSource = {}));
var IterableInAppDeleteSource;
(function (IterableInAppDeleteSource) {
    IterableInAppDeleteSource[IterableInAppDeleteSource["inboxSwipe"] = 0] = "inboxSwipe";
    IterableInAppDeleteSource[IterableInAppDeleteSource["deleteButton"] = 1] = "deleteButton";
    IterableInAppDeleteSource[IterableInAppDeleteSource["unknown"] = 100] = "unknown";
})(IterableInAppDeleteSource || (IterableInAppDeleteSource = {}));
class IterableEdgeInsets {
    top;
    left;
    bottom;
    right;
    constructor(top, left, bottom, right) {
        this.top = top;
        this.left = left;
        this.bottom = bottom;
        this.right = right;
    }
    static fromDict(dict) {
        return new IterableEdgeInsets(dict["top"], dict["left"], dict["bottom"], dict["right"]);
    }
}
class IterableHtmlInAppContent {
    type = IterableInAppContentType.html;
    edgeInsets;
    html;
    constructor(edgeInsets, html) {
        this.edgeInsets = edgeInsets;
        this.html = html;
    }
    static fromDict(dict) {
        return new IterableHtmlInAppContent(IterableEdgeInsets.fromDict(dict["edgeInsets"]), dict["html"]);
    }
}
class IterableInboxMetadata {
    title;
    subtitle;
    icon;
    constructor(title, subtitle, icon) {
        this.title = title;
        this.subtitle = subtitle;
        this.icon = icon;
    }
    static fromDict(dict) {
        return new IterableInboxMetadata(dict["title"], dict["subtitle"], dict["icon"]);
    }
}
export { IterableInAppShowResponse, IterableInAppTriggerType, IterableInAppTrigger, IterableInAppContentType, IterableEdgeInsets, IterableHtmlInAppContent, IterableInboxMetadata, IterableInAppLocation, IterableInAppCloseSource, IterableInAppDeleteSource, };
